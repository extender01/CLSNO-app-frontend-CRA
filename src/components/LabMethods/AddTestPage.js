import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import LabMetForm from './LabMetForm';
import { startAddTest } from '../../actions/testActions';


const AddTestPage = (props) => (
    <React.Fragment>
        {props.error &&   <p>{props.error.message}</p>}
        <LabMetForm new={true} formSubmit={props.startAddTest} />
    </React.Fragment>
);



const mapStateToProps = (state) => {
    return {
        error: state.labTests.error
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        startAddTest: (test) => {
            dispatch(startAddTest(test));
        }
    };
};

AddTestPage.propTypes = {
    error: PropTypes.string,
    startAddTest: PropTypes.func
};


export default connect(mapStateToProps, mapDispatchToProps)(AddTestPage);

