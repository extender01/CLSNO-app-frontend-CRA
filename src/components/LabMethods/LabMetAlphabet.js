import React from 'react';
import PropTypes from 'prop-types';

import {alphabetFilter} from '../../actions/filterActions';
import {connect} from 'react-redux';

const alphabet = ['0-9','A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','Q','R','S','T','U','V','W','X','Y','Z'];


const LabMetAlphabet = (props) => (
    <div className='fcrw lm__abc'>
        <button className={props.active === '' ? 'lm__abc--active' : undefined} name='' onClick={(e) => {props.alphabetFiltering(e.target.name)}}>VSE</button>
        {alphabet.map((item) => (
            <button className={props.active === item ? 'lm__abc--active' : undefined} key={item} name={item} onClick={(e) => {props.alphabetFiltering(e.target.name)}}>{item}</button>
       
        ))}
   
    </div>
   
);

const mapStateToProps = (state) => {
    return {
        active: state.labTests.filters.alphabet
    };
};



const mapDispatchToProps = (dispatch) => {
    return {
        alphabetFiltering: (letter) => {
            dispatch(alphabetFilter(letter));
        }
    };

};

LabMetAlphabet.propTypes = {
    active: PropTypes.string,
    alphabetFiltering: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(LabMetAlphabet);