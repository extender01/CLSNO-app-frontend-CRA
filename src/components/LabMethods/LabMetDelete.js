import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import {startDeleteTest} from '../../actions/testActions';

const LabMetDelete = (props) => (
    <div>
        <button onClick={() => {props.deleteTest(props.id)}}>SMAZAT METODU {props.id}</button>
    </div>
);

const mapDispatchToProps = (dispatch) => {
    return {
        deleteTest: (id_arg) => {
            dispatch(startDeleteTest(id_arg));
        }
    };
};

LabMetDelete.propTypes = {
    deleteTest: PropTypes.func,
    id: PropTypes.string
};

export default connect(undefined, mapDispatchToProps)(LabMetDelete);