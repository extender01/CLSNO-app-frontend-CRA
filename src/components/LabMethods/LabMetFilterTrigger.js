import React from 'react';

// import PropTypes from 'prop-types';

//nested destructuring
const LabMetFilterTrigger = (props) => (
    <div className='lm__collapsible__filters__heading'>
        <p>{props.heading}</p>
    </div>
);


// LabMetFilterTrigger.propTypes = {
    
//     }) 
// };

export default LabMetFilterTrigger;  