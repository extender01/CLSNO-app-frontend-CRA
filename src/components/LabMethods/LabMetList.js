import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import LabMetItem from './LabMetItem';
import labMetSelector from '../../selectors/labMetSelector';

const LabMetList = (props) => (
    <div className='lm__methods'>
        
    
        {
            props.labTests.length === 0 ? (
                <React.Fragment>
                    <h2>Laboratorn√≠ metody nenalezeny</h2>
                </React.Fragment>
            ) : (
                props.labTests.map((labTest) => {
                    return <LabMetItem key={labTest._id} {...labTest} />;
                })
            )
        }
    </div>

);

const mapStateToProps = (state) => {
    return {
        labTests: labMetSelector(state.labTests.tests, state.labTests.filters),
    };
};

LabMetList.propTypes = {
    labTests: PropTypes.arrayOf(PropTypes.object)
};

export default connect(mapStateToProps)(LabMetList);

