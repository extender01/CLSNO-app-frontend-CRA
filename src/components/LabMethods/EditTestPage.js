import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';


import LabMetForm from './LabMetForm';
import LabMetDelete from './LabMetDelete';
import { startEditTest } from '../../actions/testActions';

//When clicked on submit button, formSubmit method is called in LabMetForm, with current component state object as parameter. That state object gets passed here as formData parameter...
//..for onClickEditFunction which calls startEditTest with parameters of id (found in redux based on id in url - by function in mapStateToProps, we could use id from URL directly...
//but we would not know if test in redux exist, this way we do) and data from form. This function makes dispatch action to save updates to db and update redux as well 

class EditTestPage extends React.Component {
    
    onClickEdit = (formData) => {
        console.log('formData', formData);
        this.props.startEditTest(this.props.particularTest._id, formData) 
    }

    render() {
        return (
            <React.Fragment>
                <LabMetForm new={false} labMetToEdit={this.props.particularTest} formSubmit={this.onClickEdit}  />
                {this.props.particularTest ? <LabMetDelete id={this.props.particularTest._id}/> : <p>ahoj</p>}
            </React.Fragment>
        );
    }
}


const mapStateToProps = (state, ownProps) => {
    return {
        particularTest: state.labTests.tests.find((item) => item._id === ownProps.match.params.id),
        loading: state.labTests.loading,
        error: state.labTests.error
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        startEditTest: (id_arg, test_arg) => {
            dispatch(startEditTest(id_arg, test_arg));
        }
    };
};

EditTestPage.propTypes = {
    startEditTest: PropTypes.func,
    particularTest: PropTypes.object,
};


export default connect(mapStateToProps, mapDispatchToProps)(EditTestPage);
