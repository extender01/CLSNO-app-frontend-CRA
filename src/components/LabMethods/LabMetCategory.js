import React from 'react';
import { connect } from 'react-redux';
import Collapsible from 'react-collapsible';
import PropTypes from 'prop-types';

import {categoryFilter, statimFilter, erFilter, additionalFilter, clearFilters, groupFilter} from '../../actions/filterActions';
import LabMetFilterTrigger from './LabMetFilterTrigger';
import LabMetAlphabet from './LabMetAlphabet';

const LabMetCategory = (props) => (
    <React.Fragment>
        
        <div className='lm__cat'>
            <button className={props.active === 'all' ? 'lm__cat-active': undefined} name='all' onClick={() => {props.clearAllFilters()}} >VŠECHNY</button>
            <button className={props.active === 'internal' ? 'lm__cat-active' : undefined} name='internal' onClick={(e) => {props.categoryFilter(e.target.name)}} >INTERNÍ</button>
            <button className={props.active === 'external' ? 'lm__cat-active' : undefined} name='external' onClick={(e) => {props.categoryFilter(e.target.name)}} >EXTERNÍ</button>

        </div>
        <Collapsible trigger='Další filtry'>
            <Collapsible trigger={<LabMetFilterTrigger heading='PROVOZ' />} classParentString='lm__collapsible__filters'>


                <div className='lm__cat'>
                    <button className={props.statim === true ? 'lm__cat-active' : undefined} name='stat' onClick={() => {props.statimFilter(!props.statim)}} >STATIM</button>
                    <button className={props.er === true ? 'lm__cat-active' : undefined} name='er' onClick={() => {props.erFilter(!props.er)}} >POHOTOVOST</button>
                    <button className={props.additional === true ? 'lm__cat-active' : undefined} name='additional' onClick={() => {props.additionalFilter(!props.additional)}} >DOHLASKA</button>

                </div>

            </Collapsible>

            <Collapsible trigger={<LabMetFilterTrigger heading='KATEGORIE A SKUPINY' />} classParentString='lm__collapsible__filters'>


                <div className='lm__cat'>
                    <button className={props.groups === 'zaklad' ? 'lm__cat-active' : undefined} name='groups' onClick={() => groupSelector(props, 'zaklad')} >Základní panel</button>
                    <button className={props.groups === 'jatra' ? 'lm__cat-active' : undefined} name='groups' onClick={() => groupSelector(props, 'jatra')} >Vyšetření jater</button>
                    <button className={props.groups === 'ledviny' ? 'lm__cat-active' : undefined} name='groups' onClick={() => groupSelector(props, 'ledviny')}>Vyšetření ledvin</button>

                </div>

            </Collapsible>    
            
            
            <Collapsible trigger={<LabMetFilterTrigger heading='ABECENDĚ' />} classParentString='lm__collapsible__filters'>
                <LabMetAlphabet />
            </Collapsible>
        </Collapsible>

    
    </React.Fragment>
);


const mapStateToProps = (state) => {
    return {
        active: state.labTests.filters.category,
        statim: state.labTests.filters.statim,
        er: state.labTests.filters.er,
        additional: state.labTests.filters.additional,
        groups: state.labTests.filters.groups
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        categoryFilter: (category) => {
            dispatch(categoryFilter(category));
        },
        clearAllFilters: () => {
            dispatch(clearFilters());
            dispatch(erFilter(false));
            dispatch(additionalFilter(false));
            dispatch(statimFilter(false));
        },
        statimFilter: (statim) => {
            dispatch(statimFilter(statim));
            dispatch(erFilter(false));
            dispatch(additionalFilter(false));
        },
        erFilter: (er) => {
            dispatch(erFilter(er));
            dispatch(statimFilter(false));
            dispatch(additionalFilter(false));

        },
        additionalFilter: (additional) => {
            dispatch(additionalFilter(additional));
            dispatch(statimFilter(false));
            dispatch(erFilter(false));
        },
        groupFilter: (group) => {
            dispatch(groupFilter(group));
        }
       

    };
};

// checks if there is applied group filter in redux state (in props via mapStateToProps) and if its equal to passed value (keyword, that means that same button for group was clicked second time) - if that is true filter is reset to ''
//if passed keyword is different (group button clicked for first time or after DIFFERENT button, then filer is set to that keyword)
// props are passed because this fuction is outside of component and doesnt have access via HOC from connect redux
const groupSelector = (props, keyword) => {
    if(props.groups === keyword) {
        props.groupFilter('');
    } else {
        props.groupFilter(keyword);
    }
};

LabMetCategory.propTypes = {
    active: PropTypes.string,
    clearAllFilters: PropTypes.func,
    categoryFilter: PropTypes.func,
    statim: PropTypes.bool,
    er: PropTypes.bool,
    additional: PropTypes.bool,
    statimFilter: PropTypes.func,
    erFilter: PropTypes.func,
    additionalFilter: PropTypes.func,
    groupFilter: PropTypes.func
};


export default connect(mapStateToProps, mapDispatchToProps)(LabMetCategory)

