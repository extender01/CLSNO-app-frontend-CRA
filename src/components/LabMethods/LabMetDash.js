import React from 'react';
import {connect} from 'react-redux';

import LabMetList from './LabMetList';
import LabMetCategory from './LabMetCategory';
import LabMetSearch from './LabMetSearch';
import {startLoadTests} from '../../actions/testActions';




class LabMetDash extends React.Component {
    constructor(props) {
        super(props);
        
        

    }

    componentDidMount() {
        // this.props.loadTests();
        

    }

    render() {
        return (
            <div className='lm__container'>
                <div className='lm'>
                    {this.props.labTestsError === null ?
                        <React.Fragment>
                            <LabMetSearch />
                            <LabMetCategory />
                            <LabMetList />
                        </React.Fragment>
                        : 
                        <h2>nelze se spojit s db</h2>}
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        labTestsError: state.labTests.error
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        loadTests: () => {
            dispatch(startLoadTests());
        }
    };
};


export default connect(mapStateToProps, mapDispatchToProps)(LabMetDash);


