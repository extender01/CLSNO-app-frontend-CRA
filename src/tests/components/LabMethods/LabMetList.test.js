import React from 'react';
import {shallow} from 'enzyme';
import {LabMetList} from '../../../components/LabMethods/LabMetList';
import {labmets} from '../../fixtures/labmets';

test('should render LabMetList with lab methods', () => {
    const wrapper = shallow(<LabMetList labTests={labmets}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render LabMetList with warning, that there are no methods', () => {
    const wrapper = shallow(<LabMetList labTests={[]} />);
    expect(wrapper).toMatchSnapshot();
});