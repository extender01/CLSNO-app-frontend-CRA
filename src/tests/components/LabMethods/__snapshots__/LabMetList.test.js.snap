// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LabMetList with lab methods 1`] = `
<div
  className="lm"
>
  <Connect(LabMetItem)
    __v={4}
    _id="5be55cf769cf385290c66948"
    additionalOrder="24h"
    bioHalfLife="12h"
    categories=""
    category="internal"
    customNotes={
      Array [
        Object {
          "_id": "5bf40e1b60ba09405472cebd",
          "customNote": "sd",
          "department": "s1",
        },
      ]
    }
    draw="sg, se, pli, custom: je treba nejaka specialita"
    erCare={false}
    erTime="5h"
    expertise="801"
    extHowOften=""
    extResponse=""
    groupAk="CL zaklad bio"
    interfereDown="koureni"
    interfereUp="huleni"
    key="5be55cf769cf385290c66948"
    metodics="UV-VIS"
    name="ALP"
    nameAk="S_ALP"
    note="zadna poznamka"
    parcelNote=""
    parcelPreanal=""
    parcelWho=""
    physVar="velka"
    preanal="standard"
    refRange={
      Array [
        Object {
          "age": "0-150",
          "range": "0.22-0.60",
          "sex": "female",
        },
        Object {
          "age": "554",
          "range": "14+7",
          "sex": "female",
        },
        Object {
          "age": "15-50",
          "range": "40-60",
          "sex": "male",
        },
      ]
    }
    responseNote="odpoledne to flakame"
    rutCare={true}
    rutTime="5h"
    statCare={false}
    statTime="1h"
    syn="alkalicka fosfataza, hodne dalsich slov pro synonyma"
    unit="ukat/l"
    whenExtTransport=""
    where="SNO"
  />
  <Connect(LabMetItem)
    __v={0}
    _id="5bea74c7c75591445c168693"
    additionalOrder=""
    bioHalfLife=""
    categories=""
    category="external"
    customNotes={Array []}
    draw="sg"
    erCare={false}
    erTime="5h"
    expertise="801"
    extHowOften="denne"
    extResponse="3 dny"
    groupAk="EXT_VSE"
    interfereDown=""
    interfereUp=""
    key="5bea74c7c75591445c168693"
    metodics="CLIA"
    name="CYFRA"
    nameAk="S_EXT_CYFRA"
    note="baf"
    parcelNote="nesrat se s tim"
    parcelPreanal="stocit stahnout"
    parcelWho="RIA"
    physVar=""
    preanal="standard"
    refRange={Array []}
    responseNote=""
    rutCare={true}
    rutTime=""
    statCare={false}
    statTime="1h"
    syn="cyp je to"
    unit="UI/ml"
    whenExtTransport="po-pa"
    where="Krnov"
  />
  <Connect(LabMetItem)
    __v={10}
    _id="5bec433f51eacf1980555d86"
    additionalOrder="24h"
    bioHalfLife="2h"
    categories=""
    category="internal"
    customNotes={
      Array [
        Object {
          "_id": "5bf6f73c50230716143bf56d",
          "customNote": "kuzelkz",
          "department": "s1",
        },
      ]
    }
    draw="pli"
    erCare={false}
    erTime="5h"
    expertise="801"
    extHowOften=""
    extResponse=""
    groupAk="CL zaklad"
    interfereDown="kren"
    interfereUp="horcice"
    key="5bec433f51eacf1980555d86"
    metodics="CLIA"
    name="BNP"
    nameAk="P_BNP"
    note="musi byt plasma"
    parcelNote=""
    parcelPreanal=""
    parcelWho=""
    physVar="velka"
    preanal="standard"
    refRange={
      Array [
        Object {
          "age": "0-15 let",
          "range": "0 - 148",
          "sex": "Male",
        },
      ]
    }
    responseNote="kdyz pozde tak druhy den"
    rutCare={true}
    rutTime="5h"
    statCare={false}
    statTime="1h"
    syn="NT-proBNP"
    unit="pg/ml"
    whenExtTransport=""
    where="SNO"
  />
</div>
`;

exports[`should render LabMetList with warning, that there are no methods 1`] = `
<div
  className="lm"
>
  <h2>
    Laboratorn√≠ metody nenalezeny
  </h2>
</div>
`;
